schematic/store/synapse.py

```diff
@@ -1724,12 +1720,19 @@ def getDatasetProject(self, datasetId: str) -> str:
             str: The Synapse ID for the parent project.
         """
 
-        self._query_fileview()
-
         # Subset main file view
         dataset_index = self.storageFileviewTable["id"] == datasetId
         dataset_row = self.storageFileviewTable[dataset_index]
 
+        # re-query if no datasets found
+        if len(dataset_row) == 0:
```

Comment:
Is `dataset_row` a df?
If so could you do something like `if dataset_row.empty` and `if not dataset_row.empty`?

Reply:
@mialy-defelice yes; I could use `dataset_row.empty` in place of this line. As far as using `if not dataset_row.empty`; it seems like there's a reason why it's checking for exactly one, as opposed to just any, but I'm not completely sure why so I'm hesitant to change it right now